apis/rabbitmq/v1beta1/rabbitmqpolicy_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RabbitMQPolicySpec defines the desired state of RabbitMQPolicy
+type RabbitMQPolicySpec struct {
+	// +kubebuilder:validation:Required
+	// RabbitmqClusterName - the name of the RabbitMQ cluster
+	RabbitmqClusterName string `json:"rabbitmqClusterName"`
+
+	// +kubebuilder:validation:Optional
+	// VhostRef - reference to the RabbitMQVhost resource (defaults to "/")
```

Comment:
do or should we create a default vhost CR which reflects "/" for a cluster? of maybe "/" is not really a vhost and therefore does not make sense?